azurelinuxagent/common/exception.py

```diff
@@ -210,6 +210,19 @@ def __init__(self, msg=None, inner=None):
         super(ResourceGoneError, self).__init__(msg, inner)
 
 
+class InvalidExtensionEventError(AgentError):
+    """
+    Error thrown when the extension telemetry event is invalid as defined per the contract with extensions.
```

Comment:
How do you handle malformed events?

Reply:
Exceptions are thrown in the extension_telemetry.py file and caught and logged in the GAGenericLogs table. Does that answer your question? I might have misunderstood it

Reply:
I noticed there wasn't a defined error for a malformed event and you already differentiate in detail between these errors, so wondering why that is.

Reply:
I'm sorry I didnt quite get your question. If the event file is malformed, I throw the `InvalidExtensionEventError` along with the specific error code as defined below. These mainly cover the instances where the malformed part of the extension event is the breaking of the contract that we set with them. Apart from this, if there are any other malformed events (like bad json for example), we have a blanket catch for it and report it accordingly. Did that answer your question? If not we can sync offline about it

Reply:
Everything is clear once I re-read the code that reads and parses the events, so there's definitely a blanket catch for this which is the thing I was worried about. All good!